source/common/quic/envoy_quic_client_connection.cc

```diff
@@ -172,10 +172,10 @@ void EnvoyQuicClientConnection::probeWithNewPort(const quic::QuicSocketAddress&
   Network::Address::InstanceConstSharedPtr new_local_address;
   if (current_local_address->ip()->version() == Network::Address::IpVersion::v4) {
     new_local_address = std::make_shared<Network::Address::Ipv4Instance>(
-        current_local_address->ip()->addressAsString());
+        current_local_address->ip()->addressAsString(), &current_local_address->socketInterface());
```

Comment:
I'm not sure that I understand this change. Can you say more about it? I can't quite plumb it all through in my mind.

Reply:
This is not quite relevant to the rest of the PR. The Ipv4Instance has a handle to a socket interface extension which will be used to create a specific IoHandle instance.  If not passed in, the global default extension will be use. 

Here we are creating an alternative socket for this connection. The alternative socket should use the same socket interface extension to create IoHandle. Since the address cache is specific to IoSocketHandleImpl, we want to make sure the same extension is used across the connection life time.